packages/psss-server/src/connections/MeditrakConnection.ts

```diff
@@ -3,10 +3,112 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import { generateId } from '@tupaia/database';
+import { convertPeriodStringToDateRange } from '@tupaia/utils';
 import { ApiConnection } from './ApiConnection';
 
 const { MEDITRAK_API_URL = 'http://localhost:8090/v2' } = process.env;
 
+const SURVEY_RESPONSE_FIELDS = <const>['entity.code', 'survey.code', 'submission_time', 'id'];
+const ANSWER_FIELDS = <const>['question.code', 'id'];
+const PSSS_SURVEY_RESPONSE_ANSWER_TYPE = 'Number'; // All survey response answers are 'Number' in PSSS
+
+type SurveyResponseObject = {
+  [key in typeof SURVEY_RESPONSE_FIELDS[number]]: string;
+};
+
+type AnswerObject = {
+  [key in typeof ANSWER_FIELDS[number]]: string;
+};
+
 export class MeditrakConnection extends ApiConnection {
   baseUrl = MEDITRAK_API_URL;
+
+  async findSurveyResponse(surveyCode: string, orgUnitCode: string, period: string) {
+    const [startDate, endDate] = convertPeriodStringToDateRange(period);
+    const results = (await this.get(`surveyResponses/`, {
+      page: '0',
+      pageSize: '1',
+      columns: `["${SURVEY_RESPONSE_FIELDS.join('","')}"]`,
+      filter: JSON.stringify({
+        'survey.code': { comparisonValue: surveyCode, castAs: 'text' },
+        'entity.code': { comparisonValue: orgUnitCode, castAs: 'text' },
+        submission_time: {
+          comparator: 'BETWEEN',
+          comparisonValue: [startDate, endDate],
+          castAs: 'date',
+        },
+      }),
+      sort: '["submission_time DESC"]',
```

Comment:
Nice, so if there happens to be more than one it will update the most recent.

Reply:
Unfortunately, I think this may be an area where we will struggle with the whole `endTime` vs `submissionTime` thing. So in theory we want to be working with the most recent `endTime` (ie. the most recently edited) survey response. However, since the `report-server` has no access to that, it can only return the most recent `submissionTime`. I figured it's more important for this endpoint to be editing the survey response that the user is seeing data for in PSSS, but it does sorta mean that a survey response that comes through via meditrak which doesn't have a more recent `submission_time` will kinda get lost in limbo 🤷

Reply:
Oh man, I was caught out by my own terrible name here 🤦 

Yep all good, there's another card that end_time stuff is deferred into right?

Reply:
Yup: https://github.com/beyondessential/tupaia-backlog/issues/1691